src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+import { CloudFormation } from '../../shared/cloudformation/cloudformation'
+import { openLaunchJsonFile } from '../../shared/sam/debugger/commands/addSamDebugConfiguration'
+import { recordSamOpenConfigUi } from '../../shared/telemetry/telemetry.gen'
+
+const localize = nls.loadMessageBundle()
+
+export function registerSamInvokeVueCommand(context: ExtContext): vscode.Disposable {
+    return vscode.commands.registerCommand(
+        'aws.lambda.invokeForm',
+        async (launchConfig?: AwsSamDebuggerConfiguration) => {
+            await createVueWebview<SamInvokerRequest, SamInvokerResponse>({
+                id: 'createLambda',
+                name: localize('AWS.command.lambda.editor.title', 'SAM Debug Configuration Editor'),
+                webviewJs: 'samInvokeVue.js',
+                onDidReceiveMessageFunction: async (message, postMessageFn, destroyWebviewFn) =>
+                    handleFrontendToBackendMessage(message, postMessageFn, destroyWebviewFn, context),
+                context: context.extensionContext,
+                cssFiles: ['samInvokeForm.css'],
+                initialCalls: launchConfig
+                    ? [
+                          {
+                              command: 'loadSamLaunchConfig',
+                              data: {
+                                  launchConfig: launchConfig,
+                              },
+                          },
+                      ]
+                    : undefined,
+            })
+
+            recordSamOpenConfigUi()
+        }
+    )
+}
+
+export interface SamInvokeVueState {
+    launchConfig: MorePermissiveAwsSamDebuggerConfiguration
+    payload: { value: string; errorMsg: string }
+}
+
+export interface MorePermissiveAwsSamDebuggerConfiguration extends AwsSamDebuggerConfiguration {
+    invokeTarget: {
+        target: 'template' | 'api' | 'code'
+        templatePath: string
+        logicalId: string
+        lambdaHandler: string
+        projectRoot: string
+    }
+}
+
+export interface LoadSamLaunchConfigResponse {
+    command: 'loadSamLaunchConfig'
+    data: {
+        launchConfig: AwsSamDebuggerConfiguration
+    }
+}
+
+export interface GetSamplePayloadResponse {
+    command: 'getSamplePayload'
+    data: {
+        payload: string
+    }
+}
+
+export interface GetTemplateResponse {
+    command: 'getTemplate'
+    data: {
+        template: string
+        logicalId: string
+    }
+}
+
+export interface SamInvokerBasicRequest {
+    command: 'loadSamLaunchConfig' | 'getSamplePayload' | 'getTemplate'
+}
+
+export interface SamInvokerLaunchRequest {
+    command: 'saveLaunchConfig' | 'invokeLaunchConfig'
+    data: {
+        launchConfig: AwsSamDebuggerConfiguration
+    }
+}
+
+export type SamInvokerRequest = SamInvokerBasicRequest | SamInvokerLaunchRequest
+export type SamInvokerResponse = LoadSamLaunchConfigResponse | GetSamplePayloadResponse | GetTemplateResponse
```

Comment:
the subtypes could be defined inline here, they are not used anywhere else. Giving them names implies they have more meaning.

Reply:
I forget if there was a reason I didn't use those types in the functions called by the below switch/case. Let me see if that works? Then we can propagate the types to those individual functions.